backend/FwLite/MiniLcm/Validators/EntryValidator.cs

```diff
@@ -7,7 +7,39 @@ public class EntryValidator : AbstractValidator<Entry>
 {
     public EntryValidator()
     {
+        RuleFor(e => e.DeletedAt).Null();
+        RuleFor(e => e.LexemeForm).Required();
+        RuleFor(e => e.CitationForm).NoEmptyValues();
+        RuleFor(e => e.LiteralMeaning).NoEmptyValues();
+        RuleFor(e => e.Note).NoEmptyValues();
         RuleForEach(e => e.Senses).SetValidator(entry => new SenseValidator(entry));
-        //todo just a stub as an example for senses
+        RuleForEach(e => e.Components).Must(NotBeComponentSelfReference);
+        RuleForEach(e => e.Components).Must(HaveCorrectComponentEntryReference);
+        RuleForEach(e => e.ComplexForms).Must(NotBeComplexFormSelfReference);
+        RuleForEach(e => e.ComplexForms).Must(HaveCorrectComplexFormEntryReference);
+        // RuleForEach(e => e.Components).SetValidator(entry => new ComplexFormComponentValidator(entry)); // TODO: Not implemented yet
+        // RuleForEach(e => e.ComplexForms).SetValidator(entry => new ComplexFormComponentValidator(entry)); // TODO: Not implemented yet
+        // TODO: ComplexFormComponentValidator(entry) might need to know the "direction" of the entry it's validating, i.e. one class for "I'm a component" and another for "I'm the complex entry"
+        RuleForEach(e => e.ComplexFormTypes).SetValidator(new ComplexFormTypeValidator());
+    }
+
+    private bool NotBeComponentSelfReference(Entry entry, ComplexFormComponent component)
+    {
+        return component.ComponentEntryId != entry.Id;
```

Comment:
this should not be `Guid.Empty` otherwise it won't make sense

Reply:
I tried adding a `.When()` condition to this check, but FluentValidation's `.When` takes a `Func<Entry, bool>` and there isn't a `WhenForEach`. So I'd either have to use a `.ForEach` with a `.When`, or I could just make these predicates pass when the GUID in question is empty. I chose the latter approach in commit 87b2f8b3.

Reply:
looks like this may have gotten mixed up with the `HaveCorrectComponentEntryReference` I missed it on my first pass too. This check needs to prevent the Id from being empty, as it can't be inferred from the parent. It needs to be not the parent and and not empty, otherwise we don't know what you're referencing.